draft-ietf-privacypass-key-consistency.md

```diff
@@ -91,59 +99,97 @@ solutions to this problem.
+
+# Consistency Requirements {#reqs}
 
 Privacy-focused protocols which rely on widely shared public keys typically
-require keys be consistent and correct. Informally, key consistency is the
-requirement that all users who communicate with an entity share the same view
-of the key associated with that entity; key correctness is that the key's
-secret information is controlled by the intended entity and is not known to be
-available to an external attacker.
-
-Some protocols depend on large sets of users with consistent keys for privacy
-reasons. Specifically, all users with a consistent key represent an anonymity
-set wherein each user of the key in that set is indistinguishable from the
-rest. An attacker that can actively cause inconsistent views of keys can
-therefore compromise user privacy.
-
-An attacker that can cause a user to use an incorrect key will likely compromise
-the entire protocol, not just privacy.
-
-Reliant systems must also consider agility when trying to satisfy these requirements. A naive solution to
-ensuring consistent and correct keys is to only use a single, fixed key pair for the entirety of
-the system. Users can then embed this key into software or elsewhere as needed, without any additional
-mechanics or controls to ensure that other users have a different key. However, this solution clearly
+require keys be consistent. Informally, key consistency is the
+requirement that all clients who use a source-provided key in some reliant system
+share the same view of the key. Some protocols depend on large sets of clients
+with consistent keys for privacy reasons. Specifically, all clients with a
+consistent key represent an anonymity set wherein each client of the key in
+that set is indistinguishable from the rest. An attacker that can actively
+cause inconsistent views of keys can therefore compromise client privacy.
+
+## Consistency Definitions
+
+Formally, consistency is a predicate defined based on key sets. Typically, clients
+try to assess consistency of one key against one or more keys, but there are
+no restrictions on whether the clients holding those keys are the same.
+
+There are two different predicates for consistency, defined below.
+
+- Consistency: Two key sets with the same set ID are consistent iff the intersection
+  of key IDs for each set is non-empty.
+- Global consistency: A key set is globally consistent iff, for all key sets with the
+  same set ID, the set of key IDs intersects with the key IDs of the candidate key set.
+
+Checking for consistency or global consistency of two sets of key sets (singletons or not)
+consists in applying a verification function to those sets. If the two sets are consistent
+and the union of those two sets is equal to the set of all possible honestly generated values,
+then the union is globally consistent.
+
+Consistency checks can happen within a reliant system or out of it. We refer to these
```

Comment:
Having trouble parsing this. I get in-band (it's part of the protocol) and out of band (it's a separate check that a client initiates), but the description here is more opaque.